src/mpl/src/gpu/mpl_gpu_cuda.c

```diff
@@ -74,3 +74,27 @@ int MPL_gpu_ipc_close_mem_handle(void *ptr)
   fn_fail:
     return MPL_ERR_GPU_INTERNAL;
 }
+
+int MPL_gpu_malloc_host(void **ptr, size_t size)
```

Comment:
I'm thinking about add `int id` as one more parameter here to select GPU device.`
`device id` is queried from `MPL_gpu_query_pointer_type`.

Same for `free_host` below.

@gcongiu @sagarth @yfguo comments?

Reply:
Do we need to register memory with a specific device? I guess that might be important for ZE?

Reply:
To support L-0, we'd need an extra parameter for device.

Reply:
@sagarth can answer ZE part.
My thought above is for multiple-GPUs support.
We need to switch gpu id to different gpu, right?
`cudaSetDevice ( int device )`

Reply:
Also, `cudaPointerAttributes` from `MPL_gpu_query_pointer_type` provides device id alreay.

Reply:
I don't see a device parameter here:

```
ze_result_t zeDriverAllocHostMem(ze_driver_handle_t hDriver, constze_host_mem_alloc_desc_t *host_desc, size_t size, size_t alignment, void **pptr);
```

Reply:
I don't think this is required for ZE (or CUDA).  Maybe there's some confusion for host-buffer vs. device buffer allocation going on.

Reply:
What about multiple-GPUs, an example of allocating pinned host memory for different GPU?
I thought CUDA always needs to call `cudaSetDevice ( int device )` to switch different GPU.
@pavanbalaji

Reply:
Not for host memory.  Host memory is not assigned to any device.  In Ze, AFAICT, there's not even an API to do what you are suggesting.

Reply:
That's right. Forgot to notice that this was `host` malloc. We don't need device for host memory.

Reply:
I see. Thanks for clarification. I thought it is per-device pinned memory.

Reply:
Okay, thanks guys. Keep this interface not changed.